src/main/java/io/github/queerbric/inspecio/tooltip/SpawnEntityTooltipComponent.java

```diff
@@ -59,17 +59,21 @@ public SpawnEntityTooltipComponent(InspecioConfig.EntityConfig config, Entity en
 				villagerData.putString("type", "minecraft:plains");
 				itemEntityNbt.put("VillagerData", villagerData);
 			}
-			if (itemEntityNbt.contains("id", NbtElement.STRING_TYPE)) { // The spawn egg specifies its own entity type.
-				var id = itemEntityNbt.getString("id");
-				id = id.replaceAll("[^a-z0-9/._-]", "");
-				itemEntityNbt.putString("id", id);
-				Optional<EntityType<?>> specifiedEntityType = EntityType.fromNbt(itemEntityNbt);
-				if (specifiedEntityType.isPresent()) {
-					var actualEntity = specifiedEntityType.get().create(client.world);
-					if (actualEntity != null) {
-						entity = actualEntity;
-						adjustEntity(entity, itemNbt, entitiesConfig);
-					}
+			if (itemEntityNbt.contains(Entity.ID_KEY, NbtElement.STRING_TYPE)) { // The spawn egg specifies its own entity type.
+				var id = itemEntityNbt.getString(Entity.ID_KEY);
+				if (id.startsWith("minecraft:")) {
+					id = id.substring(10);
```

Comment:
I definitely don't understand the goal of this as namespaces should not have a special treatment?

Reply:
yes, but I do not see that it works when I put "minecraft:" at the beginning, I have not tested if the same thing happens with the entities of mods